instrumentation/opentelemetry-instrumentation-httpx/src/opentelemetry/instrumentation/httpx/__init__.py

```diff
@@ -731,8 +732,16 @@ def _instrument(self, **kwargs):
         self._original_async_client = httpx.AsyncClient
         request_hook = kwargs.get("request_hook")
         response_hook = kwargs.get("response_hook")
-        async_request_hook = kwargs.get("async_request_hook", request_hook)
-        async_response_hook = kwargs.get("async_response_hook", response_hook)
+        if iscoroutinefunction(request_hook):
+            async_request_hook = kwargs.get("async_request_hook", request_hook)
+        else:
+            async_request_hook = kwargs.get("async_request_hook")
+        if iscoroutinefunction(response_hook):
+            async_response_hook = kwargs.get(
+                "async_response_hook", response_hook
+            )
+        else:
+            async_response_hook = kwargs.get("async_response_hook")
         if callable(request_hook):
             _InstrumentedClient._request_hook = request_hook
         if callable(async_request_hook):
```

Comment:
And check here if it is a coroutine

Reply:
I believe you'd just need the below:
```
async_request_hook = kwargs.get("async_request_hook")

if callable(async_request_hook) and iscoroutinefunction(async_request_hook):
     _InstrumentedAsyncClient._request_hook = async_request_hook
```

Reply:
Thanks @xrmx and @lzchen for your feedback. I've tried the code you mentioned but the testAsyncInstrumentationIntegration test case will fail with the change. The reason is in this case, request_hook is actually an aysnc request hook which we cannot set to None. Please correct me if I misunderstood.